project/Settings.scala

```diff
@@ -63,14 +62,18 @@ object Settings {
     "-Ywarn-infer-any",                 // Warn when a type argument is inferred to be `Any`.
     "-Xlint:nullary-override",          // Warn when non-nullary `def f()' overrides nullary `def f'.
     "-Xlint:nullary-unit",              // Warn when nullary methods return Unit.
+    "-Xfuture",                         // Turn on future language features.
     "-opt:l:inline",
     "-opt-inline-from:<sources>"
   )
 
   private val scalacOptionsFor13 = Seq(
     "-Xlint:_",
     "-opt:l:inline",
-    "-opt-inline-from:<sources>"
+    "-opt-inline-from:<sources>",
+    "-Xlint:-byname-implicit", // See https://github.com/scala/bug/issues/12072#issuecomment-884514638
```

Comment:
This is to resolve the

```
[warn] /Users/mdedetrich/github/kanadi/src/test/scala/org/zalando/kanadi/api/EventPublishRetrySpec.scala:34:62: Block result was adapted via implicit conversion (method apply) taking a by-name parameter
[warn]     ConfigSource.default.at("kanadi.http-config").loadOrThrow[HttpConfig].copy(failedPublishEventRetry = true)
```

warnings. This is unfortunately an issue in shapeless/scalac's macros and nothing we can do about without disabling the warning entirely.